src/treetensornetworks/projttns/projttn_apply.jl

```diff
@@ -71,3 +71,34 @@ function make_environment(P::ProjTTNApply, state::AbstractTTN, e::AbstractEdge)
   )
   return P
 end
+
+function contract(P::ProjTTNApply)::ITensor
```

Comment:
I'm confused, shouldn't something like this have already been defined if `ProjTTNApply` was working before?

Reply:
The only place `ProjTTNApply` was used was in `contract_updater`. There, the argument `v` in `contract(P,v)` was constructed by accessing init_state(P). Now, this is done inside contract(P::ProjTTNApply).

Reply:
But why not keep it as it was? What does this new design improve on?